packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts

```diff
@@ -0,0 +1,45 @@
+import { NumberFieldProps } from "./number-field";
+
+export const validatePositive =
+  (props: NumberFieldProps) => (value: string) => {
+    return props.allowNegative
+      ? true
+      : Number(value) > 0
+        ? true
+        : "Value must be a positive value";
+  };
+
+export const validateIsBigInt =
+  (props: NumberFieldProps) => (value: string) => {
+    const isLargeNumber = Math.abs(Number(value)) > Number.MAX_SAFE_INTEGER;
+    return isLargeNumber && !props.isBigInt
+      ? "Value is too large for standard integer, set isBigInt to true"
+      : true;
+  };
+
+export const validatePrecision =
+  (props: NumberFieldProps) => (value: string) => {
+    if (props.precision === 0) return true;
+    if (props.precision === undefined) return true;
+    const decimalPart = value.toString().split(".")[1];
+    return decimalPart && decimalPart.length <= (props.precision ?? 0)
+      ? true
+      : `Value must have ${props.precision} decimal places`;
```

Comment:
If precision is 0 and receives a value with 1 decimal place, it should fail, right?

Reply:
good catch , done